network/src/noise/handshake.rs

```diff
@@ -249,49 +243,45 @@ impl NoiseUpgrader {
                 Some(&payload),
                 &mut client_noise_msg,
             )
-            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+            .map_err(NoiseHandshakeError::BuildClientHandshakeMessageFailed)?;
```

Comment:
why the "build" here?

Reply:
we're constructing the client message (vs sending it idk) : p